migrate/3.0-vault.sh

```diff
@@ -5,15 +5,70 @@ set -e
 function import_vault() {
     echo "...importing Vault..."
 
+    VAULT_STS="vault"
     VAULT_POD="vault-0"
+    VAULT_PVC="data-vault-0"
+    TEST_POD="ztest"
+    TEST_POD_YAML="$DIR/ztest-pod.yml"
+    VAULT_PVC_YAML="$DIR/vault-pvc.yml"
     VAULT_BU="${BACKUP_DIR}/circleci-vault"
 
-    ### Seal
-    kubectl -n "$NAMESPACE" exec "$VAULT_POD" -c vault -- vault operator seal
-    kubectl -n "$NAMESPACE" cp -v=2 -c vault "$VAULT_BU"/vault-backup.tar.gz "$VAULT_POD":/tmp/vault-backup.tar.gz
-    kubectl -n "$NAMESPACE" exec "$VAULT_POD" -c vault -- rm -rf /vault/file/*
-    kubectl -n "$NAMESPACE" exec "$VAULT_POD" -c vault -- tar -xvzf /tmp/vault-backup.tar.gz -C /vault/
-    kubectl -n "$NAMESPACE" exec "$VAULT_POD" -c vault -- rm -f /tmp/vault-backup.tar.gz
+    # Scaling down Vault
+    kubectl -n "$NAMESPACE" scale --replicas=0 sts/$VAULT_STS
+
+    # Waiting for Vault termination
+    while [[ $(kubectl -n "$NAMESPACE" get pods -l app="$VAULT_STS" 2>/dev/null | grep -c "$VAULT_STS") -gt 0 ]]
+    do
+        echo "Vault is terminating"
+        sleep 15
+    done
+
+    # Fetching the PVC info
+    VOL_SIZE=$(kubectl -n "$NAMESPACE" get pvc/$VAULT_PVC -o=jsonpath="{.status.capacity.storage}")
+    echo "Volume size of $VAULT_PVC is - $VOL_SIZE"
+
+    # Deleting the PVC
+    kubectl -n "$NAMESPACE" delete pvc/$VAULT_PVC
+
+    # Recreating the PVC
+    sed  "s/VOLUME_SIZE/$VOL_SIZE/" < "$VAULT_PVC_YAML" | kubectl -n "$NAMESPACE" apply -f -
+    kubectl -n "$NAMESPACE" get pvc/$VAULT_PVC
+
+    # Creating a test pod
+    kubectl -n "$NAMESPACE" apply -f "$TEST_POD_YAML"
+
+    # Waiting for Pod ztest
+    while [[ ! $(kubectl -n "$NAMESPACE" get pods -l app="$TEST_POD" 2>/dev/null | grep -c "1/1") -gt 0 ]]
+    do
+        echo "Waiting for ztest to scale up"
+        sleep 15
+    done
+
+    kubectl -n "$NAMESPACE" get pods -l app="$TEST_POD"
+
+    # Copy the vault backup into test pod
```

Comment:
@lloydChris We don't want to have any kind of data when restoring it which is causing this issue. If we boot up the Vault, it will create some data in PVC which is kind of same situation for which we wrote this script.